Source/hydro/Castro_hydro.cpp

```diff
@@ -245,22 +245,132 @@ Castro::check_for_cfl_violation(const Real dt)
 
     Real courno = -1.0e+200;
 
-    const Real *dx = geom.CellSize();
+    auto dx = geom.CellSizeArray();
 
     MultiFab& S_new = get_new_data(State_Type);
 
+    int ltime_integration_method = time_integration_method;
+
 #ifdef _OPENMP
 #pragma omp parallel reduction(max:courno)
 #endif
     for (MFIter mfi(S_new, hydro_tile_size); mfi.isValid(); ++mfi) {
 
         const Box& bx = mfi.tilebox();
 
-#pragma gpu box(bx)
-        ca_compute_cfl(BL_TO_FORTRAN_BOX(bx),
-                       BL_TO_FORTRAN_ANYD(q[mfi]),
-                       BL_TO_FORTRAN_ANYD(qaux[mfi]),
-                       dt, AMREX_REAL_ANYD(dx), AMREX_MFITER_REDUCE_MAX(&courno), print_fortran_warnings);
+        Real* courno_d = AMREX_MFITER_REDUCE_MAX(&courno);
+
+        auto qaux_arr = qaux.array(mfi);
+        auto q_arr = q.array(mfi);
+
+        AMREX_PARALLEL_FOR_3D(bx, i, j, k,
+        {
+
+            // Compute running max of Courant number over grids
+
+            Real dtdx = dt / dx[0];
+
+            Real dtdy = 0.0_rt;
+            if (AMREX_SPACEDIM >= 2) {
+                dtdy = dt / dx[1];
+            }
+
+            Real dtdz = 0.0_rt;
+            if (AMREX_SPACEDIM == 3) {
+                dtdx = dt / dx[2];
+            }
+
+            Real courx = (qaux_arr(i,j,k,QC) + std::abs(q_arr(i,j,k,QU))) * dtdx;
+            Real coury = (qaux_arr(i,j,k,QC) + std::abs(q_arr(i,j,k,QV))) * dtdy;
+            Real courz = (qaux_arr(i,j,k,QC) + std::abs(q_arr(i,j,k,QW))) * dtdz;
+
+            if (ltime_integration_method == 0) {
+
+                // CTU integration constraint
```

Comment:
actually, this version is valid for time_integration_method = 0 or 3

Reply:
I suspect we missed this routine when we did simplified SDC.